src/lib/types.ts

```diff
@@ -1,10 +1,5 @@
-import type { ethers } from 'ethers';
-import type { Network } from '@ethersproject/networks';
+import type { BaseProvider, MetaMaskInpageProvider } from '@metamask/providers';
 
 export type Account = string;
-
-export interface ProviderStore {
-	provider?: ethers.providers.Provider;
-	accounts?: Account[];
-	network?: Network;
-}
+export type ExternalProvider = BaseProvider;
+export type Provider = MetaMaskInpageProvider & ExternalProvider;
```

Comment:
Why the `ExternalProvider` type? And why the union of `MetaMaskInpageProvider` and `ExternalProvider`?
It seems like it's unnecessary for the working of the `global.ethereum`

Reply:
Eventually we should also not call it `Provider` since we eventually will have `JSONRPCProviders`, `APIProviders`, `HW Providers` etc.
Maybe we can do something like 
```suggestion
export type BrowserInjectedProvider = MetaMaskInpageProvider;
```
The `BaseProvider` we could use directly when needed I suppose

Reply:
> t seems like it's unnecessary for the working of the `global.ethereum`

I implemented this to avoid a type error that appeared on [`.isMetaMask`](https://github.com/deckardkrug/web3-stores/blob/develop/src/routes/index.svelte#L33).

Reply:
> Eventually we should also not call it `Provider` since we eventually will have `JSONRPCProviders`, `APIProviders`, `HW Providers` etc.

Same for this. I did it to avoid type errors and to allow for a general `.conntect` method.

Reply:
Okay great :+1: